### ISSUE TITLE
ARIA-expanded lost in disabled COMBOBOX

### ISSUE THREAD

**Issue by dkireev** [tailwindlabs/headlessui#2602]
Package: "@headlessui/react": "1.7.13",
All browsers
Link to a repo's file: https://github.com/coingaming/moon-design/blob/develop/workspaces/core/src/combobox/Combobox.tsx

Description:
In your Combobox there's an Input element with `role="combobox"` which is losing `aria-expanded` attribute once I set a Combobox state to disabled. Accessibility tests are failing because `aria-expanded` is required when `role="combobox"` is used. Here's the link to a description on [developer.mozilla.org](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role).

Your function InputFn has a [line of code](https://github.com/tailwindlabs/headlessui/blob/076b03cf491629b1930531c4ca616902be6c1688/packages/%40headlessui-react/src/components/combobox/combobox.tsx#L1023 ). if changed to `'aria-expanded': data.comboboxState === ComboboxState.Open,`, that would solve the issue.

**Comment by RobinMalfait** [comment]
Hey, thank you for this bug report! 🙏

It is odd that the Combobox (or any other component) can be opened and then later becomes disabled. But in the case this happens, then you are right that the `aria-expanded` state should still be based on whether the component is open or closed.

This should be fixed by #2610, and will be available in the next release.

You can already try it using:
- `npm install @headlessui/react@insiders`.
- `npm install @headlessui/vue@insiders`.
